src/ray/core_worker/experimental_mutable_object_manager.h

```diff
@@ -113,38 +114,54 @@ class ExperimentalMutableObjectManager {
   /// \param[in] object_id The ID of the object.
   Status SetError(const ObjectID &object_id);
 
+  /// Returns a unique semaphore name for the object. This name is intended to be used for
+  /// the object's named sempahore.
+  ///
+  /// \param[in] The ID of the object.
+  std::string GetSemaphoreName(const ObjectID &object_id);
+
+  /// Returns the named semaphore for the object. `OpenSemaphore()` must be called before
+  /// this method.
+  ///
+  /// \param[in] The ID of the object.
+  sem_t *GetSemaphore(const ObjectID &object_id);
+
+  /// Opens a named semaphore for the object. This method must be called before
+  /// `GetSemaphore()`.
+  ///
+  /// \param[in] The ID of the object.
+  void OpenSemaphore(const ObjectID &object_id);
+
+  /// Closes the named semaphore for the object. Note that the destructor calls this
+  /// method for all remaining objects.
+  ///
+  /// \param[in] The ID of the object.
+  void CloseSemaphore(const ObjectID &object_id);
```

Comment:
Delete or DestroySemaphore may be a better name for this since we also delete it from the semaphore map?